play-framework/async-http/test/controllers/HomeControllerTest.java

```diff
@@ -0,0 +1,184 @@
+                + r.getStatus()
+                + " | Response: " + r.getBody() + " | Current Time:" + System.currentTimeMillis());
+              completedReqs.incrementAndGet();
+          });
+
+        log.debug(
+          "Waiting for requests to be completed. Current Time: " + System.currentTimeMillis());
+        while (completedReqs.get() != 1) {
+            Thread.sleep(100);
+        }
+        log.debug("All requests have been completed. Exiting test.");
+    }
+
+    @Test
+    public void givenMultipleRequestsWhenResponseThenLog() throws InterruptedException {
+        AtomicInteger completedReqs = new AtomicInteger(0);
+        WSClient ws = play.test.WSTestClient.newClient(port);
+        IntStream.range(0, 100)
+                 .parallel()
+                 .forEach(num ->
+                   ws.url(url)
+                     .setRequestFilter(new AhcCurlRequestLogger())
+                     .addHeader("key", "value")
+                     .addQueryParameter("num", "" + num)
+                     .get()
+                     .thenAccept(r -> {
+                         log.debug(
+                           "Thread#" + num + " Request complete: Response code = " + r.getStatus()
+                             + " | Response: " + r.getBody() + " | Current Time:"
+                             + System.currentTimeMillis());
+                         completedReqs.incrementAndGet();
+                     })
+                 );
+
+        log.debug(
+          "Waiting for requests to be completed. Current Time: " + System.currentTimeMillis());
+        while (completedReqs.get() != 100) {
+            Thread.sleep(100);
+        }
+        log.debug("All requests have been completed. Exiting test.");
+    }
+
+    @Test
+    public void givenLongResponseWhenTimeoutThenHandle() throws InterruptedException {
+        WSClient ws = play.test.WSTestClient.newClient(port);
+        Futures futures = app.injector()
+                             .instanceOf(Futures.class);
+        CompletionStage<Result> f = futures.timeout(ws.url(url)
+                                                      .setRequestTimeout(Duration.of(1, SECONDS))
+                                                      .get()
+                                                      .thenApply(result -> {
+                                                          try {
+                                                              Thread.sleep(2000L);
+                                                              return Results.ok();
+                                                          } catch (InterruptedException e) {
+                                                              return Results.status(
+                                                                SERVICE_UNAVAILABLE);
+                                                          }
+                                                      }), 1L, TimeUnit.SECONDS);
```

Comment:
This code is deeply nested to the right. I think it needs an `extract method` on the code for the `thenApply` and you need to reconsider the formatting

Reply:
I just added a return character.
Now it looks fine.